test/ServerSpec.hs

```diff
@@ -0,0 +1,112 @@
+                                                                                                                , userName = Just (ANil ,())
+                                                                                                                , userBestFriend = Nothing
+                                                                                                                , userFriends = Nothing }
+                                                                                                         )
+                                                                                    }
+                                                                             )
+                                                        }
+                                             )
+                              }
+                         )
+
+getUserTestString :: ByteString
+getUserTestString = " query { \n getUser(id: 1) { \n userId \n userName \n userBestFriend { \n userName \n } \n } \n } \n "
+
+getUserTestQuery :: Either String (Gql GqlQuery () () 'Query)
+getUserTestQuery = Right (Gql { query = Just (ANil
+                                             , GqlQuery { getAllUsers = Nothing
+                                                        , getUser = Just (Arg (Id 1) :@@ ANil
+                                                                             , User { userId = Just (Arg Nothing :@@ ANil ,())
+                                                                                    , userName = Just (ANil ,())
+                                                                                    , userBestFriend = Just (Arg Nothing :@@ ANil
+                                                                                                            , User { userId = Nothing
+                                                                                                                   , userName = Just (ANil ,())
+                                                                                                                   , userBestFriend = Nothing
+                                                                                                                   , userFriends = Nothing 
+                                                                                                                   }
+                                                                                                            )
+                                                                                    , userFriends = Nothing
+                                                                                    }
+                                                                             )
+                                                        }
+                                             )
+                              }
+                         )
+
+
+deriving instance Show (GqlQuery 'Response)
```

Comment:
Prefer the same `deriving instance AllHave Show (GqlQuery ts) => Show (GqlQuery ts)` stuff